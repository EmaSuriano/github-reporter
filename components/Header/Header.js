import React, { Component, createRef } from 'react';
import { func, string } from 'prop-types';
import { Box, ResponsiveContext, Text, TextInput } from 'grommet';
import { Search } from 'grommet-icons';
import styled from 'styled-components';
import Slide from 'react-reveal/Slide';

import { getDirection } from '../../utils/helpers';

const PageTitle = styled(Text)`
  letter-spacing: 0.3rem;
  text-transform: uppercase;
`;
export default class Header extends Component {
  static propTypes = {
    searchProfile: func.isRequired,
    query: string,
  };

  static defaultProps = {
    query: '',
  };

  inputRef = createRef();

  state = { query: this.props.query };

  componentDidMount() {
    const { query } = this.props;
    if (!query) this.inputRef.current.focus();
  }

  handleInput = e => this.setState({ query: e.target.value });

  handleSearch = () => this.props.searchProfile(this.state.query);

  handleKeyPress = e => e.key === 'Enter' && this.handleSearch();

  render() {
    const { query } = this.state;

    return (
      <Slide top>
        <ResponsiveContext.Consumer>
          {size => (
            <Box
              direction={getDirection(size)}
              align="center"
              justify="start"
              pad={{ horizontal: 'medium', vertical: 'small' }}
              gap="medium"
            >
              <Box>
                <PageTitle level={1} size="large" weight="bold">
                  Github Reporter
                </PageTitle>
              </Box>
              <Box direction="row" basis="full" align="center" gap="small">
                <Box
                  basis="full"
                  direction="row"
                  align="center"
                  pad={{ horizontal: 'small', vertical: 'xsmall' }}
                  margin="medium"
                  round="small"
                  border={{
                    side: 'all',
                    color: 'border',
                  }}
                >
                  <Search onClick={this.handleSearch} />
                  <TextInput
                    type="search"
                    ref={this.inputRef}
                    plain
                    value={query}
                    onChange={this.handleInput}
                    onKeyPress={this.handleKeyPress}
                    placeholder="Who are you looking for ?"
                  />
                </Box>
              </Box>
            </Box>
          )}
        </ResponsiveContext.Consumer>
      </Slide>
    );
  }
}
