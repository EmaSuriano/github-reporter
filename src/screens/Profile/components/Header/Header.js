import React, { Component } from "react";
import { func } from "prop-types";
import { FaSearch } from "react-icons/fa";

export default class Header extends Component {
  static propTypes = {
    searchProfile: func.isRequired
  };

  state = { query: "" };

  handleInput = e => this.setState({ query: e.target.value });

  handleSearch = () => this.props.searchProfile(this.state.query);

  render() {
    return (
      <section className="profile--header">
        <h1 className="header--title">Github Reporter</h1>
        <div className="input-search">
          <input
            type="text"
            placeholder="Who are you looking for ?"
            onChange={this.handleInput}
            value={this.state.query}
          />
          <span className="input-search--button" onClick={this.handleSearch}>
            <FaSearch size="2rem" />
          </span>
        </div>
      </section>
    );
  }
}
