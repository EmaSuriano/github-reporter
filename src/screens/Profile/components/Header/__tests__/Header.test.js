import React from "react";
import { shallow } from "enzyme";

import Header from "../Header";

const searchProfile = jest.fn();

const props = {
  searchProfile
};

describe("Header Component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Header {...props} />);
  });

  describe("render", () => {
    it("should match with html structure of the snapshot", () =>
      expect(wrapper).toMatchSnapshot());
  });

  describe("methods", () => {
    const mockedEvent = { target: { value: "wesbos" } };

    describe("handleInput", () => {
      let query;

      beforeAll(() => {
        wrapper.instance().handleInput(mockedEvent);
        query = wrapper.state().query;
      });

      it("should set state with a new value", () =>
        expect(query).toBe("wesbos"));
    });

    describe("handleSearch", () => {
      beforeAll(() => {
        wrapper.instance().handleInput(mockedEvent);
        wrapper.instance().handleSearch();
      });

      it("should call to searchProfile function", () =>
        expect(searchProfile).toHaveBeenCalledTimes(1));

      it("should call with the state", () =>
        expect(searchProfile).toHaveBeenCalledWith("wesbos"));
    });
  });
});
