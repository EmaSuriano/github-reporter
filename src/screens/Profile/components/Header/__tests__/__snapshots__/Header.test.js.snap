// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component render should match with html structure of the snapshot 1`] = `
<section
  className="profile--header"
>
  <h1
    className="header--title"
  >
    Github Reporter
  </h1>
  <div
    className="input-search"
  >
    <input
      onChange={[Function]}
      placeholder="Who are you looking for ?"
      type="text"
      value=""
    />
    <span
      className="input-search--button"
      onClick={[Function]}
    >
      <FaSearch
        size="2rem"
      />
    </span>
  </div>
</section>
`;
